modules/ROOT/pages/custom-view.adoc

```diff
@@ -0,0 +1,81 @@
+= Custom View
+:navtitle: Custom View
+:description: How to create a custom view in TinyMCE.
+:keywords: view, UI, addView
+
+{productname} allows developers to create custom views and attach them to the main editor view. This feature enables the creation of additional UI components that can be toggled on or off. When toggled on, the custom view replaces the editor, providing a distinct visual space for specialized functionality.
+
+include::partial$misc/admon-iframe-only.adoc[]
+
+== Editor View API
+
+The `addView` API in the editor's UI registry allows developers to register new view containers. These containers are initially hidden (off) and attached next to the main view. The visibility of these custom views can be toggled using the `ToggleView` command. The state of the custom view can be queried using the `queryCommandValue` method.
+
+This is the syntax for the `addView` function: `+editor.ui.registry.addView(name: String, obj: View.ViewSpec)+`
+
+=== Parameters
+
+[[name]]
+==== `+name+`
+
+The `+name+` parameter is a unique identifier for the new view.
+
+*Type:* `+String+`
+
+[[obj]]
+==== `+obj+`
+
+The `+obj+` parameter is the view specification object.
+
+*Type:* `+View.ViewSpec+`
+
+=== View Specification Object
+
+The view specification object can have various properties:
+
+[[buttons]]
+==== `+buttons+`
+
+The `+buttons+` property specifies an array of buttons to be displayed in the top right corner of the view.
+
+*Type:* `+Array+`
+
+[[onShow]]
+==== `+onShow+`
+
+The `+onShow+` property specifies a function to be called when the view is displayed. It receives an API object.
+
+*Type:* `+Function+`
+
+[[onHide]]
+==== `+onHide+`
+
+The `+onHide+` property specifies a function to be called when the view is hidden. It receives an API object.
+
+*Type:* `+Function+`
```

Comment:
@FarzadHayat - Suggest adding it into a table format for better visibility:
![Screenshot 2024-02-22 at 1 10 09 pm](https://github.com/tinymce/tinymce-docs/assets/96402951/f3ae785e-386c-4a40-ba51-f066ff393e98)

```suggestion
=== Parameters

[cols="1,1,1"]
|===
| Property | Description | Type

| `+name+` | The `+name+` parameter is a unique identifier for the new view. | `+String+`

| `+obj+` | The `+obj+` parameter is the view specification object. | `+View.ViewSpec+`
|===

=== View Specification Object

[cols="1,1,1"]
|===
| Property | Description | Type

| `+buttons+` | The `+buttons+` property specifies an array of buttons to be displayed in the top right corner of the view. | `+Array+`

| `+onShow+` | The `+onShow+` property specifies a function to be called when the view is displayed. It receives an API object. | `+Function+`

| `+onHide+` | The `+onHide+` property specifies a function to be called when the view is hidden. It receives an API object. | `+Function+`
|===
```